training/sklearn/structured/base/trainer/utils.py

```diff
@@ -35,8 +35,14 @@ def data_train_test_split(data_df):
                   pandas.DataFrame, pandas.Series)
     """
 
-    # Only use metadata.FEATURE_NAMES + metadata.TARGET_NAME
-    features = data_df[metadata.FEATURE_NAMES]
+    if metadata.FEATURE_NAMES is None:
```

Comment:
Does it ever happen ? I see metadata.FEATURE_NAMES declared as constant.

Reply:
Yes, metadata.py is one file we suggest users to change according to their needs. So, it might not happen in this sample, but since we are providing a template, it might happen in other samples (as a matter of fact, it has a different value in the base sample).